docs/design/auditing-containment.md

```diff
@@ -0,0 +1,168 @@
+# Auditing and Node Containment
+
+## Abstract
+
+This design doc outlines how we will implement "containment mode," in which the auditing service monitors a node that has accepted an audit request, but refuses to send the requested data.
+Currently in the codebase, nodes that do this will be marked as having audit failures.
+However, we don't want to mark nodes as offline or immediately mark an audit failure if this specific case arises.
+Instead, we want to "contain" these nodes by retrying the original audit until they either eventually respond and pass the audit, or refuse to respond a certain number of times and ultimately receive an audit failure mark.
+
+## Main Objectives
+1. Nodes won’t be able to “escape” audits for specific data by refusing to send requested data.
+2. The audit service should not audit contained nodes for other data, and should verify their originally requested data.
+3. The overlay service should not select contained nodes.
```

Comment:
Is it just upload node selection, or download/checker/repairer as well?

Reply:
Following our conversation this morning, it seems like it's fine for contained nodes to be selected (only the audit service should care about contained status), so I'll remove this objective.